tests/php/src/Commission/OrderRefundCommissionTest.php

```
@@ -0,0 +1,342 @@
+		$order  = new WC_Order( $order->get_id() );
+		$refund = $this->create_refund( $order, 15 );
+
+		$refund_commission = $this->get_refund_commission( $refund, $order )->calculate();
+
+		// Default: no gateway returns its fee, nothing is deducted.
+		$this->assertEquals( 0.0, round( $refund_commission->get_vendor_gateway_fee_refund(), 2 ) );
+		$this->assertEquals( 0.0, round( $refund_commission->get_admin_gateway_fee_refund(), 2 ) );
+		$this->assertEquals( 13.5, round( $refund_commission->get_vendor_total_refund(), 2 ) ); // 27/30 × 15
+
+		// A gateway that returns its fee (e.g. Paystack) supplies the amount.
+		$supply_prorated_fee = function ( $gateway_fee_refund, $prorated_fee ) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$gateway_fee_refund'. (undefined)

(UnusedFormalParameter)